packages/location-management/src/components/LocationTagAddEdit/tests/Form.test.tsx

```diff
@@ -0,0 +1,216 @@
+import { notification } from 'antd';
+import flushPromises from 'flush-promises';
+import fetch from 'jest-fetch-mock';
+import { store } from '@opensrp/store';
+
+import Form, { onSubmit } from '../Form';
+import * as fixtures from './fixtures';
+import { act } from 'react-dom/test-utils';
+
+describe('containers/pages/locations/Form', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('renders without crashing', () => {
+    const wrapper = mount(
+      <Provider store={store}>
+        <Router history={history}>
+          <Form />
+        </Router>
+      </Provider>
+    );
+    expect(wrapper.props()).toMatchSnapshot();
+  });
+
+  it('tests cancel button', () => {
+    const wrapper = mount(
+      <Provider store={store}>
+        <Router history={history}>
+          <Form />
+        </Router>
+      </Provider>
+    );
+    wrapper.find('button#cancel').simulate('click');
+  });
+
+  it('tests Create New Payload', async () => {
+    const wrapper = mount(
+      <Provider store={store}>
+        <Router history={history}>
+          <Form />
+        </Router>
+      </Provider>
+    );
+
+    // with values test
+    wrapper
+      .find('input[name="name"]')
+      .simulate('change', { target: { name: 'name', value: 'Name213' } });
+    wrapper.update();
+    wrapper
+      .find('textarea[name="description"]')
+      .simulate('change', { target: { name: 'description', value: 'this is description' } });
+    wrapper.update();
+    wrapper.simulate('submit');
+
+    await act(async () => {
+      await flushPromises();
+      wrapper.update();
+    });
+
+    wrapper.unmount();
+  });
+
+  it('tests Update Payload', async () => {
+    fetch.once(JSON.stringify(fixtures.sampleLocationTagPayload));
+    const wrapper = mount(
+      <MemoryRouter initialEntries={[`/1`]}>
+        <Provider store={store}>
+          <Route path={'/:id'} component={() => <Form id="1" />} />
+        </Provider>
+      </MemoryRouter>
+    );
+
+    await act(async () => {
+      await flushPromises();
+      wrapper.update();
+    });
+
+    expect(fetch.mock.calls).toEqual([
+      [
+        'https://opensrp-stage.smartregister.org/opensrp/rest/location-tag/1',
+        {
+          headers: {
+            accept: 'application/json',
+            authorization: 'Bearer null',
+            'content-type': 'application/json;charset=UTF-8',
+          },
+          method: 'GET',
+        },
+      ],
+    ]);
+
+    wrapper
+      .find('input[name="name"]')
+      .simulate('change', { target: { name: 'name', value: 'Name213' } });
+    wrapper.update();
+    wrapper
+      .find('textarea[name="description"]')
+      .simulate('change', { target: { name: 'description', value: 'this is description' } });
+    wrapper.update();
+    wrapper.simulate('submit');
+
+    await act(async () => {
+      await flushPromises();
+      wrapper.update();
+    });
+
+    expect(fetch.mock.calls).toEqual([
```

Comment:
I think it would be a better practice if  we tested each call separately in the array to avoid repetition. e.g `fetch.mock.calls[0]` in the first one and `fetch.mock.calls[1]` in the second one

Reply:
i do agree with you 🤔💭

Reply:
updating the tests